src/main/src/main/java/org/geoserver/catalog/util/ReaderDimensionsAccessor.java

```diff
@@ -565,6 +565,8 @@ public boolean hasResolution(String domain) {
                 Class<?> type = Class.forName(typeName);
                 if (type == java.util.Date.class) {
                     result.addAll(new TimeParser().parse(value));
+                } else if ((type == java.lang.Double.class || type == java.lang.Float.class) && !value.contains(",")) {
```

Comment:
What if it's an integer, or short, or ... whatever? Can it check Number.class.isAssignableFrom(...) instead?